contrib/upload-kktix-ticket-csv-to-bigquery.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+#
+# export GOOGLE_APPLICATION_CREDENTIALS="<where to access service-account.json>"
+#
+root_pycon_etl=${HOME}/work-my-projects/pycontw-projects/PyCon-ETL
+root_upload_data=${HOME}/work-my-projects/pycontw-projects/PyCon-ETL-working
+project_id="pycontw-225217"
+cmd_upload=${root_pycon_etl}/contrib/upload-kktix-ticket-csv-to-bigquery.py
+
+
+for year in 2018 2019 2020
```

Comment:
hardcode year? How about passing year as a arg?

Reply:
oh, I got it, because those canonical columns in that python script is dedicated to 2017, 18, 19 right? It's not a general-purpose upload script

Reply:
Is it possible to keep kktix format always the same? Otherwise, it's hard to compare data by year

Reply:
Which part of the format? Did you mean by the columns?

If you meant the columns, no, the names of the columns were manually decided at the very beginning when the ticket selling page was created. The names of the columns are managed by the registration team. `upload-kktix-ticket-csv-to-bigquery.py` is meant to make the names consistent as much as possible.

Maintaining `upload-kktix-ticket-csv-to-bigquery.py` annually is a must if we would like to keep the consistency, or we can use `upload-kktix-ticket-csv-to-bigquery.py` to upload the data if the naming convention of column is not an issue for you.

Reply:
yes I mean by the [columns](https://github.com/pycontw/PyCon-ETL/blob/master/contrib/upload-kktix-ticket-csv-to-bigquery.py#L28-L137)

> The names of the columns are managed by the registration team
If we keep the names always the same, then maintaining `column dictionary` in `upload-kktix-ticket-csv-to-bigquery.py` annually wouldn't be a must right?

Reply:
> yes I mean by the [columns](https://github.com/pycontw/PyCon-ETL/blob/master/contrib/upload-kktix-ticket-csv-to-bigquery.py#L28-L137)
> 
> > The names of the columns are managed by the registration team
> 
> If we keep the names always the same, then maintaining `column dictionary` in `upload-kktix-ticket-csv-to-bigquery.py` annually wouldn't be a must right?

Yes.

The only concern is that it is very hard to ask the registration team to keep using the same column names every year, because use cases are very likely to differ every year.

The raw names of the columns are actually the description of the ticket field. For example, the column `how_did_you_know_pycon_tw` is converted from `How did you find out PyCon TW? / 如何得知 PyCon TW` (via the `column dictionary`. `How did you find out PyCon TW? / 如何得知 PyCon TW` is created by the registration team according to real world use cases, and the description may change every year. If we would like to make sure `How did you find out PyCon TW? / 如何得知 PyCon TW` is used every year.

Summary:
In order to keep the column names the same every year in bigquery tables, we have:
- solution A -  by sending a sponsor to watch the description policy in the registration team
    - pros: we don't need to maintain the column dictionary
    - cons: a sponsor is expensive. The communication overhead is huge. (Who is willing to do this?)
solution B - maintaining the `column dictionary` (in a smart way if possible)
    - pros: we don't need a sponsor
    - cons: we need to maintain the `column dictionary`

Reply:
thanks for the summarization 😄 

I go for sol. A (btw, what's `description policy`?)